rocky/rocky/views/finding_list.py

```diff
@@ -46,13 +53,53 @@ def generate_findings_metadata(
     return sort_by_severity_desc(findings_meta)
 
 
-class FindingListView(BreadcrumbsMixin, SeveritiesMixin, OctopoesView, ListView):
-    template_name = "findings/finding_list.html"
-    paginate_by = 20
+class FindingListFilter(OctopoesView, ConnectorFormMixin, SeveritiesMixin, ListView):
+    connector_form_class = ObservedAtForm
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.severities = self.get_severities()
+        self.valid_time = self.get_observed_at()
+        self.muted_findings = request.GET.get("muted_findings", "all")
+        self.finding_types = request.GET.getlist("finding_types", [])
+        self.show_muted = False
+        self.exclude_muted = False
 
     def get_queryset(self) -> FindingList:
-        severities = self.get_severities()
-        return FindingList(self.octopoes_api_connector, self.get_observed_at(), severities)
+        if self.muted_findings == "show":
+            self.show_muted = True
+        if self.muted_findings == "exclude":
+            self.exclude_muted = True
+        return FindingList(
+            self.octopoes_api_connector,
+            self.valid_time,
+            self.severities,
+            exclude_muted=self.exclude_muted,
+            show_muted=self.show_muted,
```

Comment:
`show_muted=(self.muted_findings in ("show", "exclude"))`

You can then remove the 4 lines above the return.